### DISCUSSION TITLE
Use of tiled layouts with EBSD.plot command in MTEX 6.0.3 beta

### DISCUSSION THREAD

**Discussion by gmcastroguiza** [mtex-toolbox/mtex#2233] [Ask Anything] [answered]
Hi Everybody,

Does anybody know how to use the tiledlayout command of Matlab with the plot command of EBSD? 

I am trying to create a mosaic of 6 EBSD graphs, distributed in three groups of an IPF map and a phase map, for three experimental conditions. However, each time I start the figure, the first instance of the plot command erases all the settings I have configured on the tiledlayout command. 

Kind regards,

Gabriel Castro

**Comment by Tijmenvermeij** [comment]
There is multiple ways I think, but I usually use `nextAxis` and define the layout as well, as follows, for example:
```matlab
figure;
newMtexFigure('layout',[3 2]);

plot(ebsd1)
nextAxis;
plot(ebsd2)
nextAxis;
etc
```

Best,
Tijmen

**Reply by gmcastroguiza** [reply]
Thank you for your answer, I tried with that but the arrangements that I can make with the nextAxis instruction are not suited for what I wanted. I ended up using the following code that I was able to assemble with many suggestions by Matlab AI Playground, until I found a way that it worked for me. It is a little cumbersome though, so if anyone knows a more concise way to replicate the same arrangement, please let me know.
```matlab
cKParent=ipfHSVKey(CS{3});
cKParent.inversePoleFigureDirection=vector3d.Z;
figure (1)
Fig(1)=newMtexFigure;
t1=tiledlayout(1,4,'TileSpacing','tight');
title(t1, 'Clean up and filtering process evolution of the quenched dataset');
tile1_1=tiledlayout(t1,1,2,'TileSpacing','none');
tile1_1.Layout.Tile=1;
ax1_1=nexttile(tile1_1);
hold on
colorParentRaw=cKParent.orientation2color(rawEbsd(CS{3}.mineral).orientations);
plot(rawEbsd(CS{3}.mineral),colorParentRaw,'showMicronBar','on','parent',ax1_1);
cKChild=ipfHSVKey(CS{2});
cKChild.inversePoleFigureDirection=vector3d.Z;
colorChildRaw=cKChild.orientation2color(rawEbsd(CS{2}.mineral).orientations);
ax=gca;
hold(ax,"on");
plot(rawEbsd(CS{2}.mineral),colorChildRaw,'showMicronBar','on','hold','ax','parent',ax1_1);
hold off
nexttile(tile1_1);
hold on
plot(rawEbsd,'showMicronBar','on')
title(tile1_1,'Raw EBSD')
hold off
% _____________________________________________________________________________________________________________________
% Unfiltered data plots
tile1_2=tiledlayout(t1,1,2,'TileSpacing','none');
tile1_2.Layout.Tile=2;
nexttile(tile1_2);
colorParentUnfiltered=cKParent.orientation2color(unfilteredEbsd(CS{3}.mineral).orientations);
plot(unfilteredEbsd(CS{3}.mineral),colorParentUnfiltered,'showMicronBar','on');
hold on
colorChildUnfiltered=cKChild.orientation2color(unfilteredEbsd(CS{2}.mineral).orientations);
plot(unfilteredEbsd(CS{2}.mineral),colorChildUnfiltered,'showMicronBar','on');
hold off
nexttile(tile1_2);
plot(unfilteredEbsd,'showMicronBar','on')
title(tile1_2,'Cleaned up EBSD')
% _____________________________________________________________________________________________________________________
% Processed data plots
tile1_3=tiledlayout(t1,1,2,'TileSpacing','none');
tile1_3.Layout.Tile=3;
nexttile(tile1_3);
colorParentProcessed=cKParent.orientation2color(processedEbsd(CS{3}.mineral).orientations);
plot(processedEbsd(CS{3}.mineral),colorParentProcessed,'showMicronBar','on');
hold on
colorChildProcessed=cKChild.orientation2color(processedEbsd(CS{2}.mineral).orientations);
plot(processedEbsd(CS{2}.mineral),colorChildProcessed,'showMicronBar','on');
hold off
nexttile(tile1_3);
plot(processedEbsd,'showMicronBar','on')
title(tile1_3,'Filtered EBSD')
% _____________________________________________________________________________________________________________________
% Orientation color key
tile1_4=tiledlayout(t1,4,1,'TileSpacing','none');
tile1_4.Layout.Tile=4;
nexttile(tile1_4,3);
plot(cKParent)
```
Kind regards,

Gabriel Castro

**Reply by ralfHielscher** [reply, marked as answer]
Hi Gabriel,
actually MTEX `nextAxis` was out before Matlab introduced `tiledLayout`. This is the reason both concepts are not really compatible. In a future release of MTEX I surely will drop the MTEX implementation and go with the Matlab one. 
Currently we can only assist with using `nextAxis`.
Ralf.

**Reply by gmcastroguiza** [reply]
Hi Prof. Hielscher,
Thank you for your answer, I understand. If it helps with the development in some way, the way I produced the layout needed to create a newMtexFigure before setting the tiled layout, and to assign the parent property in each plot - perhaps except in the color key - for avoiding that the newMtexFigure command reshaped the entire figure when I tried to build the first plot. The parent property gave me some problems in other instances, since it conflicted with the implementation of the plot command, specially when I tried to plot a numeric value (that creates a color bar) in a grain dataset.
Kind regards,
Gabriel.
